text/0510-dynamodb-global-table.md

```diff
@@ -0,0 +1,1258 @@
+  tableName: 'FooTable',
+  partitionKey: { name: 'Foo', type: AttributeType.STRING },
+  sortKey: { name: 'Bar', type: AttributeType.STRING },
+  localSecondaryIndex: [
+    {
+      indexName: 'FooTableLsi',
+      sortKey: { name: 'Foo', type: AttributeType.STRING },
+    },
+  ],
+});
+```
+
+You can add local secondary index with `addLocalSecondaryIndex` method.
+
+```typescript
+globalTable.addLocalSecondaryIndex({
+  indexName: 'FooTableLsi',
+  sortKey: { name: 'Foo', type: AttributeType.STRING },
+});
+```
+
+__NOTE:__
+
+* You need a sort key to define a local secondary index.
+* You can create up to five local secondary indexes.
+
+#### Encryption
+
+Encryption mode defines how the table, replicas and GSIs would be encrypted at rest. There are now four
+types of encryptions that are available for global tables:
+
+1. `AWS_OWNED`: This uses a KMS key for encryption that is owned by DynamoDB. This is the default
+for global tables.
+2. `AWS_MANAGED`: A KMS key is created in your account and is managed by AWS.
+3. `KEY_ARNS`: [NEW] You will provide KMS key arns for each replica. You will need to specify arns for each
+replica region. The key also needs to be in the same region as the replica.
+4. `MULTI_REGION_KEY`: [NEW] A multi region KMS key and its supporting stacks in replica regions will be
+provisioned automatically.
+
+The `encryption` mode selected remains the same for each replica. If you will like to provide KMS keys managed
+by you for each replica, then you can use the `KEY_ARNS` encryption mode and provide region specific key in
+`encryptionKey` property of each replica.
+
+```typescript
+new GlobalTable(tableStack, 'FooTable', {
+  tableName: 'FooGlobalTable',
+  partitionKey: {
+    name: 'FooHashKey',
+    type: AttributeType.STRING,
+  },
+  encryption: GlobalTableEncryption.KEY_ARNS,
+  replicas: [
+    {
+      region: 'us-west-2',
+      encryptionKey: 'FooKeyArn',
+    },
+    {
+      region: 'us-east-1',
+      encryptionKey: 'BarKeyArn',
+    },
+  ],
+});
+```
+
+__NOTE__:
+
+* Encryption mode `CUSTOMER_MANAGED` is now deprecated. You can switch to either `KEY_ARNS` or `MULTI_REGION_KEY` mode.
+
+#### Grants
+
+To use one of the replicas in your application, you can use the replica's grant methods to get the necessary permissions.
+
+The global table's `replica(region)` method will return an `ITable` reference of the replica from which you will be able
+to grant permissions.
+
+```typescript
+class FooStack extends Stack {
+  public readonly globalTable: GlobalTable;
+
+  constructor(scope: App, id: string, props: StackProps) {
+    super(scope, id, props);
+
+    this.globalTable = new GlobalTable(this, 'FooTable', {
+      tableName: 'FooGlobalTable',
+      partitionKey: {
+        name: 'FooHashKey',
+        type: AttributeType.STRING,
+      },
+      replicas: [
+        {
+          region: 'us-west-2',
+          encryptionKey: 'FooKmsKeyArn',
```

Comment:
Encryption is not really relevant for this example, so if you remove it the example will be shorter and easier to read.

Reply:
Will update this.